crates/sncast/src/helpers/fee.rs

```diff
@@ -113,14 +116,23 @@ impl FeeArgs {
                         ),
                         max_gas_unit_price: Some(max_gas_unit_price.try_into_()?),
                     },
-                    (Some(max_fee), Some(max_gas), None) => FeeSettings::Strk {
-                        max_gas: Some(max_gas.try_into_()?),
-                        max_gas_unit_price: Some(
-                            max_fee
-                                .floor_div(&NonZeroFelt::from_felt_unchecked(max_gas))
-                                .try_into_()?,
-                        ),
-                    },
+
+                    (Some(max_fee), Some(max_gas), None) => {
+                        let max_gas_unit_price =
+                            max_fee.floor_div(&NonZeroFelt::from_felt_unchecked(max_gas));
```

Comment:
The unchecked word should signal that it is not something we should use, as handling it incorrectly can lead to undefined behavior. In Rust, unchecked constructs like this are typically marked as `unsafe`. For example, see [`std::num::NonZero`](https://github.com/rust-lang/rust/blob/3378a5e084fa51c2e7f5c3f53b87e8ad89f1c517/library/core/src/num/nonzero.rs#L397). I’m not sure why Jonathan didn’t make it `unsafe`.

Previously, this functionality was clearly incorrect because we didn’t check whether it was zero. Now, we do check it, but the check happens *after* the operation. 

The sane approach would look something like this:

```rust
&NonZeroFelt::try_from(max_gas_unit_price)
    .context("--max-gas calculated from --max-fee should be greater than 0. Please increase --max-fee")?
```

Additionally, please remove the other places where `from_felt_unchecked` is used.

Reply:
Also (probably) the best way to handle this situation is to create our own wrapper type with the `FromStr` trait, which disallows the number `0`. This way, `clap`, while parsing the command, would immediately return an error:

```rs
struct NonZeroFelt(Felt);

impl FromStr for NonZeroFelt {
    type Err = String;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        match s.parse::<Felt>() {
            Ok(felt) => {
                if felt == Felt::ZERO {
                    Err(String::from("value should be greater than 0"))
                } else {
                    Ok(NonZeroFelt(felt))
                }
            }
            Err(_) => Err(String::from("value should be a number")),
        }
    }
}
```

Reply:
Moved here https://github.com/foundry-rs/starknet-foundry/pull/2796 as the code was also moved here